packages/lesswrong/lib/vulcan-lib/utils.ts

```diff
@@ -324,14 +324,15 @@ const footnoteAttributes = [
   'data-footnote-section',
   'data-footnote-back-link',
   'data-footnote-back-link-href',
+  'data-internal-id'
 ]
 
 export const sanitize = function(s: string): string {
   return sanitizeHtml(s, {
     allowedTags: sanitizeAllowedTags,
     allowedAttributes:  {
       ...sanitizeHtml.defaults.allowedAttributes,
-      '*': footnoteAttributes,
+      '*': [...footnoteAttributes, 'data-internal-id'],
```

Comment:
In theory could this be restricted to `blockLevelElements` (unless I'm misunderstanding what elements might potentially have a `data-internal-id`)?

Reply:
It could, although that would be a bit annoying to implement here because you would have to add it into every individual block level element (and there's not actually a definite list of what counts as block level from ckeditor, because elements in the data view can be upcasted to block/non-block elements in the model based on custom plugin logic).

Having needed to make changes in this file a couple of times recently (for the cta button and now google doc importing), it has turned out to be a bit of a footgun generator, because it silently strips things out of the html that you try to put in there. This has led me to think it's better to be overpermissive